modal/_container_entrypoint.py

```diff
@@ -441,20 +553,110 @@ def import_function(
     else:
         obj = None
 
+    if function:
+        code_deps = function.deps(only_explicit_mounts=True)
+
     return ImportedFunction(
         obj,
         user_defined_callable,
-        function_def.webhook_config,
         active_app,
-        is_async,
-        is_generator,
-        data_format,
-        input_concurrency,
-        function_def.is_auto_snapshot,
-        function,
+        code_deps,
+    )
+
+
+def import_class_service(
+    function_def: api_pb2.Function,
+    ser_cls,
+    ser_fun,
+    ser_params: Optional[bytes],
+    client: Client,
+) -> Service:
+    """
+    This imports a full class to be able to execute any @method or webhook decorated methods.
+
+    See import_function.
+    """
+    module: Optional[ModuleType] = None
+    cls: Optional[Type] = None
+    active_app: Optional[_App] = None
+    code_deps: Optional[List["modal.object._Object"]] = None
+
+    if function_def.definition_type == api_pb2.Function.DEFINITION_TYPE_SERIALIZED:
+        assert ser_cls is not None
+        cls = ser_cls
+        # This is a serialized function we already fetched from the server
+        method_partials = _find_partial_methods_for_cls(cls, _PartialFunctionFlags.all())
+    else:
+        # Load the module dynamically
+        module = importlib.import_module(function_def.module_name)
+        qual_name: str = function_def.function_name
+
+        if not is_global_object(qual_name):
+            raise LocalFunctionError("Attempted to load a class defined in a function scope")
+
+        parts = qual_name.split(".")
+        if not (
+            len(parts) == 2 and parts[1] == "*"
+        ):  # the "function name" of a class service "function placeholder" is expected to be "ClassName.*"
+            raise InvalidError(f"Invalid 'class function' identifier {qual_name}")
```

Comment:
Should we try to reserve `InvalidError` for when the *user* does something invalid (that's how we document it)? AFAICT this is more of an internal error check?

Reply:
👍